src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -523,4 +527,23 @@ public EnumActionResult processRightClick(EntityPlayer player, net.minecraft.wor
         }
     }
 
+    private final Deque<CauseStackManager.StackFrame> tryHarvestBlockStackFrames = new LinkedList<>();
+
+    /**
+     * Some mods (e.g. OreExcavator) will call tryHarvestBlock directly, and
+     * the cause stack may not have the player on it. This should fix that.
+     * See SpongeForge#2789.
+     */
+    @Inject(method = "tryHarvestBlock", at = @At("HEAD"))
+    private void onTryHarvestBlockHead(BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
```

Comment:
with the ongoing cleanup/splitting of mixins, can you name this `impl$addPlayerToCauseStackForHarvesting`?